internal/exec/exec.go

```diff
@@ -554,7 +557,9 @@ func (e *fieldExec) execField(ctx context.Context, r *request, f *query.Field, r
 	result, err := e.execField2(spanCtx, r, f, resolver, span)
 
 	if err != nil {
-		r.addError(errors.Errorf("%s", err))
+		queryError := errors.Errorf("%s", err)
+		queryError.ResolverError = err
+		r.addError(queryError)
 		addResult(f.Alias, nil) // TODO handle non-nil
```

Comment:
The spec says that `null` should be returned on error: http://facebook.github.io/graphql/#sec-Errors-and-Non-Nullability

Actually the `null` should even propagate to the parent if the current field has a non-null type. This is not yet implemented.